src/OpenTelemetry.Api/Logs/LoggerOptions.cs

```diff
@@ -62,7 +63,45 @@ public LoggerOptions(InstrumentationScope instrumentationScope)
     /// <summary>
     /// Gets the domain of events emitted by the instrumentation library.
     /// </summary>
-    public string? EventDomain { get; init; }
+    public string? EventDomain
+    {
+        get
+        {
+            var attributes = this.InstrumentationScope.AttributeBacking;
+            if (attributes != null && attributes.TryGetValue("event.domain", out var eventDomain))
+            {
+                return eventDomain as string;
+            }
+
+            return null;
+        }
+
+        init
+        {
+            var attributes = this.InstrumentationScope.AttributeBacking;
+
+            Dictionary<string, object> newAttributes = new(attributes?.Count + 1 ?? 1);
+
+            if (attributes != null)
+            {
+                foreach (var kvp in attributes)
+                {
+                    newAttributes.Add(kvp.Key, kvp.Value);
+                }
+            }
+
+            if (value != null)
+            {
+                newAttributes["event.domain"] = value;
+            }
+            else
+            {
+                newAttributes.Remove("event.domain");
```

Comment:
move `event.domain` to a const somewhere? I guess ultimately should probably be part of the auto-generated semantic conventions.

Reply:
Updated